build/common/installer/scripts/tomlparser.rb

```diff
@@ -239,8 +239,19 @@ def populateSettingValuesFromConfigMap(parsedConfig)
         if !parsedConfig[:log_collection_settings][:metadata_collection][:include_fields].nil?
           puts "config::Using config map setting for kubernetes metadata include fields"
           include_fields = parsedConfig[:log_collection_settings][:metadata_collection][:include_fields]
-          if include_fields.kind_of?(Array)
-            @logKubernetesMetadataIncludeFields = include_fields.join(",")
+          if include_fields.empty?
+            puts "config::Include fields specified for Kubernetes metadata is empty, disabling Kubernetes metadata"
+            @logEnableKubernetesMetadata = false
+          elsif include_fields.kind_of?(Array)
+            include_fields.map!(&:downcase)
+            predefined_fields = @logKubernetesMetadataIncludeFields.downcase.split(',')
+            any_field_match = include_fields.any? { |field| predefined_fields.include?(field) }
```

Comment:
This is good, we should also log warn message if customer provided any field which is not supported. For example,  customer provider podlabels,podannotations, deploymentName.